workitem/workitem_repository.go

```diff
@@ -95,6 +97,43 @@ func (r *GormWorkItemRepository) Load(ctx context.Context, ID string) (*app.Work
 	return convertWorkItemModelToApp(wiType, res)
 }
 
+// LoadTopWorkitem returns top most work item of the list. Top most workitem has the Highest order.
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LoadTopWorkitem(ctx context.Context) (*app.WorkItem, error) {
+	res := WorkItem{}
+	r.db = r.db.Debug()
+	tx := r.db.Order("execution_order desc").Last(&res)
+	if tx.RecordNotFound() {
+		return nil, nil
+	}
+	if tx.Error != nil {
+		return nil, errors.NewInternalError(tx.Error.Error())
+	}
+	wiType, err := r.witr.LoadTypeFromDB(ctx, res.Type)
+	if err != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	return convertWorkItemModelToApp(wiType, &res)
+}
+
+// LoadBottomWorkitem returns bottom work item of the list. Bottom most workitem has the lowest order.
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LoadBottomWorkitem(ctx context.Context) (*app.WorkItem, error) {
+	res := WorkItem{}
+	tx := r.db.Order("execution_order").Last(&res)
```

Comment:
@aslakknutsen 

[1] I have created a issue for Limiting Order Of Execution within a space -> https://github.com/almighty/almighty-core/issues/972

[2] Regarding second optimization, the reason that I am reading the whole workitem and not just the execution_order is because I am interested in the workitem ID also -> to handle the case where the request is made more than once. Do you think a better way would be return ID as well as execution_order?